bc_obps/reporting/tests/service/test_report_emission_allocation_service.py

```diff
@@ -0,0 +1,855 @@
+            source_type__json_key="sourceTypeWithUnitAndFuel",
+            has_unit=True,
+            has_fuel=True,
+        )
+
+        report_activity = self.test_infrastructure.make_report_activity()
+        report_source_type = make(
+            ReportSourceType,
+            activity_source_type_base_schema=activity_source_type,
+            source_type=activity_source_type.source_type,
+            report_activity=report_activity,
+            report_version=self.test_infrastructure.report_version,
+            json_data={"test_report_source_type": "yes"},
+        )
+        report_fuel = make(
+            ReportFuel,
+            report_source_type=report_source_type,
+            report_version=self.test_infrastructure.report_version,
+            json_data={"test_report_unit": True},
+            report_unit=None,
+        )
+        report_fuel_excluded = make(
+            ReportFuel,
+            fuel_type_id=64,
+            report_source_type=report_source_type,
+            report_version=self.test_infrastructure.report_version,
+            json_data={"test_report_unit": True},
+            report_unit=None,
+        )
+
+        report_emission_1 = make(
+            ReportEmission,
+            report_fuel=report_fuel,
+            report_source_type=report_source_type,
+            report_version=self.test_infrastructure.report_version,
+            json_data={"equivalentEmission": 25.9998},
+        )
+
+        report_emission_2 = make(
+            ReportEmission,
+            report_fuel=report_fuel,
+            report_source_type=report_source_type,
+            report_version=self.test_infrastructure.report_version,
+            json_data={"equivalentEmission": 275.0001},
+        )
+        report_emission_excluded = make(
+            ReportEmission,
+            report_fuel=report_fuel_excluded,
+            report_source_type=report_source_type,
+            report_version=self.test_infrastructure.report_version,
+            json_data={"equivalentEmission": 100.0000},
+        )
+
+        report_emission_1.emission_categories.set([self.FLARING_CATEGORY_ID])
+        report_emission_2.emission_categories.set([self.FLARING_CATEGORY_ID])
+        report_emission_excluded.emission_categories.set([self.WOODY_BIOMASS_CATEGORY_ID])
+
+        report_product_1 = make(
+            ReportProduct,
+            id=3,
+            report_version=self.test_infrastructure.report_version,
+            facility_report=self.test_infrastructure.facility_report,
+            product_id=1,  #  "BC-specific refinery complexity throughput"
+        )
+        report_product_2 = make(
+            ReportProduct,
+            id=4,
+            report_version=self.test_infrastructure.report_version,
+            facility_report=self.test_infrastructure.facility_report,
+            product_id=29,  # "Sugar: solid"
+        )
+
+        self.mock_get_response = {
```

Comment:
@Sepehr-Sobhani @pbastia any thoughts on ways we could possibly cut down on the size of the mocked get repsonse / post payloads in this file? This service returns/ingests a pretty large object

Reply:
maybe a class or function to generate the data objects not being tested (ie: where allocated emissions are 0)?

Reply:
@dleard I think this is something that will be improved when we decouple the service return types from the API schemas. Using the API data types here forces us to use those large JSON objects, whereas we could use data structures much simpler or easy to manipulate for the service use case